hal/audio_extn/cirrus_sony.c

```diff
@@ -363,7 +363,16 @@ void spkr_prot_init(void *adev, spkr_prot_init_config_t spkr_prot_init_config_va
                        &handle.spkr.checksum, true);
     if (ret)
         return;
-#endif
+
+    handle.spkl.cal_ok = true;
+    handle.spkr.cal_ok = true;
+
+    int spkl_empty = 0;
+    for (i = 0; i < 4; i ++) {
+      spkl_empty |= handle.spkl.cal_r[i];
+    }
+    handle.is_stereo = spkl_empty != 0;
```

Comment:
So `spkr` is primary, and `spkl` is optional? The mixers determine this by having a separate `L` and `R` suffix (or no suffix at all for _mono_), but for the calibration I guess it makes sense when the earpiece - the left channel! - is not powered by a Cirrus DSP.